### ISSUE TITLE
Inconsistent `_lazy_values` dict once the dataclass's attribute is set

### ISSUE THREAD

**Issue by addyess** [gtsystem/lightkube#44]
When trying to update a Deployment's `topologySpreadConstraints` I unearthed a situation where an object of type `lightkube.models.core_v1.PodSpec` had a value set for its attribute `topologySpreadConstraints` and a value in in `_lazy_values` set as 

```
{"topologySpreadConstraints": None}
```

whats worse is that it wasn't consistent and debugging the issue might be hiding it.   Whatever it is... it seems that when setting a value for an attribute, the `_lazy_values` shadow value should be removed but isn't.  This resulted in the de-serialization of the Deployment into a json blob to be sent to the API didn't include these `topologySpreadConstraints`

As a workaround i'm doing this:

```python
        obj.spec.template.spec._lazy_values.pop("topologySpreadConstraints", None)
        obj.spec.template.spec.topologySpreadConstraints = [
            ... whatever i'd like here ...
        ]
````

but it feels pretty gross

**Comment by addyess** [comment]
I wrote this unit test which demonstrates it a bit simpler:

```python
from dataclasses import dataclass
from typing import List

from lightkube.core.dataclasses_dict import DataclassDictMixIn


@dataclass
class ListItem(DataclassDictMixIn):
    key: "str"
    operator: "str"
    values: "List[str]" = None


@dataclass
class ListCollection(DataclassDictMixIn):
    matchExpressions: "List[ListItem]" = None


def test_lazy_values():
    a = ListCollection.from_dict({})
    assert a.to_dict() == {}
    a.matchExpressions = [ListItem("abc", "def")]
    assert a.to_dict() == {}  #   WHAT?  Where's my matchExpressions
    a._lazy_values.pop("matchExpressions", None)
    assert a.to_dict() == dict(matchExpressions=[{"key": "abc", "operator": "def"}])
```

**Comment by gtsystem** [comment]
Thanks for bringing up the issue. As another temporary workaround (before the fix is released), you could also disable the usage of the lazy class instantiation using `client = Client(lazy=False)`.

**Comment by gtsystem** [comment]
Fixed in version 0.13.0.
